ansible/roles/kubernetes/tasks/main.yml

```diff
@@ -2,22 +2,65 @@
 - include: fedora.yml
   when: ansible_distribution == "Fedora"
 
+- name: Update Kubernetes directories if this is CoreOS
+  set_fact:
+    bin_dir: "/opt/bin"
+    kube_script_dir: "/opt/bin/kubernetes"
+  when: ansible_lsb.id == "CoreOS" and kube_script_dir == "/usr/libexec/kubernetes"
+
 - name: Update {{ kube_script_dir }} if this is atomic
   set_fact:
     kube_script_dir: "/usr/local/libexec/kubernetes"
   when: is_atomic and kube_script_dir == "/usr/libexec/kubernetes"
 
+- name: Determine if Kubernetes download directory exists
```

Comment:
Why are you checking if things exist and creating them if they don't? Seems like you are trying to make the declarative nature of ansible imperative...